### DISCUSSION TITLE
How to make standalone executables of an MTEX function

### DISCUSSION THREAD

**Discussion by TomasManik** [mtex-toolbox/mtex#847] [Ask Anything]
Hei,

has anybody done this? I would like to compile a standalone executable application of a matlab code involving some MTEX functions, e.g. plotting of the ODF. Any advice will be appreciated!

Tomas

**Comment by ralfHielscher** [comment]
If you have the Matlab compiler this should be possible.

Ralf.

**Reply by TomasManik** [reply]
Yes, I have the Matlab compiler. I tried to compile the following code:

```matlab
function texture(filename)

cs = crystalSymmetry('m-3m', [4.04 4.04 4.04], 'mineral', 'Al');
ssO = specimenSymmetry('orthorhombic');

data = readmatrix(filename,'FileType','text','NumHeaderLines',3);

degree = pi/180;

ori = [orientation.byEuler(data(1,1)*degree, data(1,2)*degree, data(1,3)*degree, cs, ssO)];
for i=2:size(data,1)
    ori(i) = orientation.byEuler(data(i,1)*degree, data(i,2)*degree, data(i,3)*degree, cs, ssO);
end
odf = calcDensity(ori,'halfwidth',5*degree);

levelsODF = [0.5, 1, 2, 5, 10, 20, 25];
odf.SS = ssO;

figure
plot(odf, 'sections', 18, 'contourf', levelsODF, 'colorrange',[0 30], 'layout',[5 4], 'coordinates','off','xlabel','','ylabel','')
mtexColorbar
```

I got the following error:

![bilde](https://user-images.githubusercontent.com/29653279/110759288-bb377a80-824d-11eb-9979-99ab3df77343.png)

Could you try if you sucseed in compiling the function above?

With best regards
Tomas

**Reply by ralfHielscher** [reply]
I do not have a Matlab compile - hence I can not test this. Have you read #492 ? Maybe you can ask there.

Ralf.

**Comment by cbetoIng** [comment]
Hi Tomas,

did you resolve your problem? I have a similar issue, with Matlab showing the same warning after compilation. The problem seems to be related to the "data" folder path as the executable crashes with the error "Data package not installed!". 

I'm compiling my exe using Application Builder...

Regards,
Christian

**Reply by TomasManik** [reply]
Hi Christian, 

I am using hte Application Builder too! I have not proceeded further here. Do you manage to compile an ODF plotter in your Matlab?
 
Tomas

**Reply by cbetoIng** [reply]
Hi Tomas,

using my Matlab, the function compiles and executes without error...mysterious.
I will keep you up to date, if I solved the problem.

Regards,
Christian

**Reply by TomasManik** [reply]
Can you send me the function which you managed to compile succesfully on your Matlab? Which Matlab and MTEX version did you use?

**Reply by cbetoIng** [reply]
Hi Tomas,

sorry for the late reply. I just copied your source code to an m-script and added it to the Application Compiler as Main-File. No problems during compilation, but during runtime, as I don't have the proper files available.

Regards,
Christian

**Reply by TomasManik** [reply]
Hi,

thank you very much for testing of compiling the script. I also managed to compile, but running it is a problem. I am sending you the proper input file to test running of the script. I very much wonder if you can run it without any error in your Matlab and Mtex. Please, let me know if you find to run it. 

Regards and thank you in advance

Tomas

[texture.zip](https://github.com/mtex-toolbox/mtex/files/6323708/texture.zip)

**Reply by cbetoIng** [reply]
By using the attatched script is was able to compile your code and plot the data. I just removed the variable `filename` from the function call, placed it in the same folder as `texture.m` and added the following line `filename = 'texture.ori';` to load the data.

I hope this works for you...

[compile_texture.zip](https://github.com/mtex-toolbox/mtex/files/6342694/compile_texture.zip)

**Reply by TomasManik** [reply]
Thank you very much! Your "compile_texture.m" worked for me! Interesting is that I dont get it work when I use Application Compiler App. I appreciate your help very much once again! :)

**Reply by xiu-shen** [reply]
Have your problems been solved? Could you please offer some suggestions?
